community/detectors/joomla_cve_2023_23752/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202323752/Cve202323752VulnDetector.java

```diff
@@ -0,0 +1,187 @@
+    private final HttpClient httpClient;
+
+    private final Clock utcClock;
+
+    @Inject
+    Cve202323752VulnDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+        this.httpClient = checkNotNull(httpClient);
+        this.utcClock = checkNotNull(utcClock);
+    }
+
+    private static StringBuilder buildTarget(NetworkService networkService) {
+        StringBuilder targetUrlBuilder = new StringBuilder();
+        if (NetworkServiceUtils.isWebService(networkService)) {
+            targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+        } else {
+            targetUrlBuilder
+                .append("http://")
+                .append(toUriAuthority(networkService.getNetworkEndpoint()))
+                .append("/");
+        }
+        return targetUrlBuilder;
+    }
+
+    @Override
+    public DetectionReportList detect(
+        TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+        logger.atInfo().log("CVE-2023-23752 starts detecting.");
+
+        return DetectionReportList.newBuilder()
+            .addAllDetectionReports(
+                matchedServices.stream()
+                    .filter(NetworkServiceUtils::isWebService)
+                    .filter(this::isServiceVulnerable)
+                    .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                    .collect(toImmutableList()))
+            .build();
+    }
+
+    private boolean isServiceVulnerable(NetworkService networkService) {
+
+        HttpHeaders httpHeaders =
+            HttpHeaders.builder()
+                .addHeader(CONTENT_TYPE, MediaType.PLAIN_TEXT_UTF_8.type())
+                .addHeader(
+                    ACCEPT,
+                    "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8")
+                .addHeader(UPGRADE_INSECURE_REQUESTS, "1")
+                .addHeader(ACCEPT_LANGUAGE, "Accept-Language: en-US,en;q=0.5")
+                .addHeader(ACCEPT_ENCODING, "gzip, deflate")
+                .build();
+
+        String targetVulnerabilityUrl = buildTarget(networkService).append(VULNERABLE_PATH).toString();
+        try {
+            HttpResponse httpResponse =
+                httpClient.send(
+                    get(targetVulnerabilityUrl).setHeaders(httpHeaders).build(), networkService);
+            if (httpResponse.status().code() != DETECTION_STRING_BY_STATUS
+                || !httpResponse.bodyString().isPresent()) {
+                return false;
+            }
+            String content_type_value = "";
+            if (httpResponse.headers().get("Content-Type").isPresent()) {
+                content_type_value = httpResponse.headers().get("Content-Type").toString();
+
+            } else if (httpResponse.headers().get("content-type").isPresent()) {
+                content_type_value = httpResponse.headers().get("Content-Type").toString();
```

Comment:
Should be lower case "content-type". Raw string value is prone to errors like this. If checking content-type is really necessary, then you could create 2 static variables for them, CONTENT_TYPE & CONTENT_TYPE_LOWER for example.